recipes/dev/lora_finetune_fsdp2.py

```diff
@@ -290,25 +291,6 @@ def _setup_model(
         with utils.set_default_dtype(self._dtype), torch.device("meta"):
             model = config.instantiate(cfg_model)
 
-        if self._is_rank_zero:
-            # The model contains LoRA params which won't have any matching keys in
-            # the state dict. As a result, we need to load with strict=False.
-            # Before loading the state dict, ensure the state dict keys for the base
-            # model and adapters (if available) match the keys in the full LoRA model
-            # This is a good sanity check to prevent silent errors
-            validate_state_dict_for_lora(
```

Comment:
`validate_state_dict_for_lora` needs some refactor since FSDP2 use clean FQNs. 
FSDP2 also check param on meta and throw Runtime error https://fburl.com/wrhxykn3

```
"FSDP parameters should be materialized from meta device before training, "
f"but the following were still on meta device: {param_names_on_meta}\n"
"For example, call module.to_empty(device) to materialize to device and "
"call module.reset_parameters() on each module to initialize values."
```

Reply:
`validate_missing_and_unexpected_for_lora` works out of the box. added it in this PR. finger acrossed!